crypto/hash/rlp.go

```diff
@@ -1,22 +1,47 @@
 package hash
 
 import (
+	"hash"
+	"sync"
+
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/rlp"
 	"golang.org/x/crypto/sha3"
 )
 
+var kec256Pool = sync.Pool{
+	New: func() interface{} {
+		return sha3.NewLegacyKeccak256()
+	},
+}
+
 // FromRLP hashes the RLP representation of the given object.
 func FromRLP(x interface{}) (h common.Hash) {
-	hw := sha3.NewLegacyKeccak256()
+	hw := kec256Pool.Get().(hash.Hash)
+	defer func() {
```

Comment:
where is this change coming from?  Is it part of upstream change? why mix it here?

Reply:
Ah, this is just to optimize the memory used for RLP hashing in multi-threading. Sorry for mixing it here. Will open another PR for this.

Reply:
Reverted and PR here: https://github.com/harmony-one/harmony/pull/3616. Please take a look.